Cython/Compiler/ExprNodes.py

```diff
@@ -1930,15 +1930,15 @@ def declare_from_annotation(self, env, as_target=False):
             return
 
         annotation = self.annotation
-        if annotation.is_string_literal:
+        if annotation.expr.is_string_literal:
             # name: "description" => not a type, but still a declared variable or attribute
```

Comment:
I wonder if this is really still true. Shouldn't string literals behave just like any other type declaration now?

Reply:
Not sure... string literals are distinguishable from other types. E.g.

```def f(a: "something"):
    pass```

gets translated to an annotation of `'"something"'` (i.e. it's quoted inside the string literal).

So I don't automatically think it makes sense to change how Cython infers the types from the annotations - it has exactly the same information as before and all that's really changed is the representation in the `__annotations__` dictionary.

With that said, this looks like a silly and inconsistent special case, so maybe it should go. I haven't done it this commit though

```def f(a: float, b: 'float'):
    c: float = a
    d: 'float' = a
    e: float
    f: 'float'```

In the above `a`, `c`, `d`, `e` end up typed as a C double, `b` ends up as a C float, and `f` ends up as a PyObject so it's all a bit odd

Reply:
Don't know if it adds anything, but it's tested in https://github.com/cython/cython/blob/20787f28fae6c0c828940a25e580b56a1f594efd/tests/run/pep526_variable_annotations.py#L40

Reply:
Your example shows that it's definitely inconsistent currently. Given PEP-563, I think strings can be ignored now, because forward declarations are no longer a problem. However, they weren't (always) ignored before, so maybe we should show a warning for them if we detect a type in a string in 3.0, and really ignore them in 3.1?

Reply:
There was actually already a warning for annotations as strings (I'm struggling to work out how to link to it here, but in this commit it's in ExprNodes.py line 13625) so I've just increased the level so it shows by default.